src/cluster.c

```diff
@@ -4390,6 +4444,83 @@ sds clusterGenNodesDescription(int filter, int use_pport) {
     return ci;
 }
 
+/* Generate a csv-alike representation of the specified cluster node.
+ * See clusterGenNodesDescription() top comment for more information.
+ *
+ * The function returns the string representation as an SDS string. */
+sds clusterGenLinkDescription(clusterLink *link) {
+    serverAssert(link);
+    char events[3], conninfo[CONN_INFO_LEN], *p;
+    sds node_name;
+
+    if (link->node) {
+        node_name = sdsnewlen(link->node->name, CLUSTER_NAMELEN);
+    } else {
+        node_name = sdsnew("unknown");
+    }
+
+    p = events;
+    if (link->conn) {
+        if (connHasReadHandler(link->conn)) *p++ = 'r';
+        if (connHasWriteHandler(link->conn)) *p++ = 'w';
+    }
+    *p = '\0';
+
+    sds ret = sdscatfmt(sdsempty(),
+        "%s %s node=%s ctime=%I events=%s sbufalloc=%U, sbufused=%U",
+        connGetInfo(link->conn, conninfo, sizeof(conninfo)),
+        link->inbound ? "from" : "to",
+        node_name,
+        link->ctime,
+        events,
+        sdsalloc(link->sndbuf),
+        sdslen(link->sndbuf));
+
+    sdsfree(node_name);
+
+    return ret;
+}
+
+/* Generate a csv-alike representation of the links ,
+ * including the "myself" node, and return an SDS string containing the
+ * representation (it is up to the caller to free it).
+ *
+ * All the nodes matching at least one of the node flags specified in
+ * "filter" are excluded from the output, so using zero as a filter will
+ * include all the known nodes in the representation, including nodes in
+ * the HANDSHAKE state.
+ *
+ * Setting use_pport to 1 in a TLS cluster makes the result contain the
+ * plaintext client port rather then the TLS client port of each node.
+ *
+ * The representation obtained using this function is used for the output
+ * of the CLUSTER NODES function, and as format for the cluster
+ * configuration file (nodes.conf) for a given node. */
+sds clusterGenLinksDescription() {
+    sds links = sdsempty(), link;
```

Comment:
If you were to call this on a single primary it would return an empty string, which I'm not sure we do anywhere. Maybe just double check that works?

Reply:
I don't think so either. Thinking about this more, the only reason CLUSTER NODEs returns a txt file is because it's mirroring the nodes.conf. Maybe we should just return this as a RESP MAP (i.e. key/value) ? That is usually more human readable any ways.

Reply:
Done in the new revision.